custom_functions/find_related_containers.py

```diff
@@ -0,0 +1,189 @@
+            containers_response = phantom.requests.get(uri=url, verify=False).json()
+            all_container_dictionary = {}
+            if containers_response['count'] > 0:
+                # Build repository of available container data
+                for data in containers_response['data']:
+                    all_container_dictionary[str(data['id'])] = data
+
+                for k,v in container_dictionary.items():
+
+                    # If any of the containers contain more than the minimum match count request that container detail.
+                    if len(container_dictionary[str(k)]) >= minimum_match_count:
+
+                        # Grab container details if its a valid container based on previous filtering.
+                        if str(k) in all_container_dictionary.keys():
+                            container_data = all_container_dictionary[str(k)]
+                            status = container_data['status']
+                            container_type = container_data['container_type']
+                            container_name = container_data['name']
+                            in_case = container_data['in_case']
+
+                            # Build final output
+                            outputs.append({'container_id': str(k),
+                                            'container_indicator_match_count': len(container_dictionary[str(k)]),
+                                            'container_status': status,
+                                            'container_type': container_type,
+                                            'container_name': container_name,
+                                            'in_case': in_case})
+            else:
+                raise RuntimeError(f"'Unable to find any valid containers at url: '{url}'")
+        elif container_number < 600 and container_number > 0:
+            phantom.debug("Fetching related container data")
+            # if the container number is smaller than 600, its faster to grab each container individiually
+            for k,v in container_dictionary.items():
+                # Dedupe the number of indicators
+                container_dictionary[str(k)] = list(set(v))
+
+                # If any of the containers contain more than the minimum match count request that container detail.
+                if len(container_dictionary[str(k)]) >= minimum_match_count:
+
+                    # Grab container details
+                    url = phantom.build_phantom_rest_url('container', k)
+                    response_data = phantom.requests.get(url, verify=False).json()
+                    status = response_data['status']
+                    container_type = response_data['container_type']
+                    container_name = response_data['name']
+                    in_case = response_data['in_case']
+
+                    # Build final output
+                    if status != filter_out_status and str(k) != str(current_container):
```

Comment:
At this point status is the name but filter_out_status could be either the name or the ID. I think in the input processing you should assign filter_out_status_name and filter_out_status_id so you can use the correct one down here.